File: KunstButikken.Frontend/src/features/art/api/art.ts
import type { ApiArt, UiArt } from '@/features/art/types/art';

export function mapApiToUi(a: ApiArt): UiArt {
  return {
    id: String(a.id ?? ''),
    titleNb: a.titleNb ?? a.titleEn ?? 'Untitled',
    titleEn: a.titleEn ?? a.titleNb ?? 'Untitled',
    descriptionNb: a.descriptionNb ?? undefined,
    descriptionEn: a.descriptionEn ?? undefined,
    artist: a.artist ?? a.sellerDisplayName ?? a.sellerId ?? 'Unknown Artist',
    sellerId: a.sellerId ?? undefined,
    sellerDisplayName: a.sellerDisplayName ?? undefined,
    price: a.price ?? 0,
    image: a.imageUrl ?? '',
    status: a.status as UiArt['status'],
    isVerified: a.isVerified ?? undefined,
    isFeatured: a.isFeatured ?? undefined,
  };
}

File: KunstButikken.ArtService/KunstButikken.ArtService/Domain/Models/Art.cs
namespace KunstButikken.ArtService.Domain.Models;

public enum ArtStatus
{
    Draft = 0,
    Published = 1,
    Sold = 2,
    Rejected = 3
}

public class Art
{
    public Guid Id { get; set; }

    // Multilingual fields
    public string TitleNb { get; set; } = string.Empty;
    public string TitleEn { get; set; } = string.Empty;
    public string DescriptionNb { get; set; } = string.Empty;
    public string DescriptionEn { get; set; } = string.Empty;

    public Uri ImageUrl { get; set; } = new("http://localhost");
    public decimal Price { get; set; }

    public Guid SellerId { get; set; }

    // The painting's credited artist (e.g. painter name)
    public string Artist { get; set; } = string.Empty;

    // Cached display name of the seller (so UI doesn't have to resolve user profiles for listings)
    public string SellerDisplayName { get; set; } = string.Empty;
    public ArtStatus Status { get; set; } = ArtStatus.Draft;
    public bool IsVerified { get; set; }
    public bool IsFeatured { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}

